quantization_troubleshooting.md

```diff
@@ -0,0 +1,153 @@
+When implementing quantization on a PyTorch model, MCT's initial step involves converting the model into a graph representation using `torch.fx`.
+However, `torch.fx` comes with certain common limitations, with the primary one being its requirement for the computational graph to remain static.
+
+Despite these limitations, some adjustments can be made to facilitate MCT quantization.
+
+**Solution**: (assuming you have access to the model's code)
+
+Check the `torch.fx` error, and search for an identical replacement:
+
+An `if` statement in a module's `forward` method might can be easily skipped.
+The `list()` Python method can be replaced with a concatenation operation [A, B, C].
+
+
+## Debugging Tools
+
+### Network Editor
+
+To pinpoint the possible cause of accuracy degradation in your model, it may be necessary to isolate the quantization settings of specific layers.
+However, this cannot be accomplished using MCT's main API.
+
+For instance, accuracy might suffer due to quantization of certain layer types, such as `Swish` for activations or `Conv2d` for weights.
+To address this issue and manipulate individual layers within the network, you can utilize the "Network Editor" API.
+
+**Implementation**:
+
+Using the network editor API (mct.core.network_editor) you can define a set of "rules" to apply on the network, using provided filters.
+
+Please refer to our [tutorial](https://github.com/sony/model_optimization/blob/main/tutorials/notebooks/keras/debug_tools/example_keras_network_editor.ipynb) for more details.
+
+A use-case example to understand if a certain layer causes accuracy degradation - set the output quantization bits of the layer's type to 16 bits (instead of the default 8 bits defined in the TPC).
+If the accuracy improves, then it is pointing to that layer for causing the issue.
+
+*Note:* This is mainly a debugging tool and should not be used as an API for quantizing a model for deployment (may result in an unstable behavior).
+
+
+## Advanced Quantization Methods
+
+Sometimes, we did everything right in the quantization process, but either the quantized model accuracy degradation is too high
+or the quantized model size is too large for our deployment target (e.g. the IMX500).
+MCT offers advanced features for handing these issues, such as mixed-precesion and GPTQ.
+
+### Mixed Precision Quantization
+
+In mixed precision quantization, MCT will assign a different number of bit widths to each weight in the model, depending on the weight's layer sensitivity
+and a constraint defined by the user, such as target model size.
+
+Check out the [mixed precision tutorial](https://github.com/sony/model_optimization/blob/main/tutorials/notebooks/keras/ptq/example_keras_mobilenet_mixed_precision.ipynb)
+for more information and an implementation example.
+
+### GPTQ - Gradient-Based Post Training Quantization
+
+When PTQ (either with or without mixed-precision) fails to deliver the required accuracy, GPTQ is potentially the remedy.
```

Comment:
mixed precision either with "-" or without. I usually write without (I saw that in papers) but just be consistent